appsec-kit-v24/src/main/java/com/vaadin/appsec/v24/views/VulnerabilitiesTab.java

```diff
@@ -0,0 +1,210 @@
+                    .equals(vulnerabilityDTO.getDependency())) {
+                return false;
+            }
+            if (vaadinAnalysisFilter != null && !vaadinAnalysisFilter
+                    .equals(vulnerabilityDTO.getVaadinAnalysis())) {
+                return false;
+            }
+            if (developerAnalysisFilter != null && !developerAnalysisFilter
+                    .equals(vulnerabilityDTO.getDeveloperStatus())) {
+                return false;
+            }
+            if (severityFilter != null && !severityFilter
+                    .equals(vulnerabilityDTO.getSeverityLevel())) {
+                return false;
+            }
+            return riskScoreFilter == null
+                    || riskScoreFilter <= vulnerabilityDTO.getRiskScore();
+        });
+    }
+
+    @Override
+    public void refresh() {
+        Set<Vulnerability> selectedItems = grid.getSelectedItems();
+        grid.deselectAll();
+        grid.setItems(AppSecService.getInstance().getVulnerabilities());
+        dependency.setItems(getListDataProvider().getItems().stream()
+                .map(Vulnerability::getDependency).collect(Collectors.toSet()));
+        applyFilters();
+        selectedItems.forEach(grid::select);
+    }
+
+    private Double getRiskScoreFromFilter(String riskScoreFilter) {
+        String[] parts = riskScoreFilter.split("=");
+        return Double.valueOf(parts[1]);
+    }
+
+    private void buildFilters() {
+        ecosystem = new ComboBox<>("Ecosystem");
+        ecosystem.setItems(Ecosystem.MAVEN, Ecosystem.NPM);
+
+        dependency = new ComboBox<>("Dependency");
+
+        vaadinAnalysis = new ComboBox<>("Vaadin analysis");
+        vaadinAnalysis.setItems(AssessmentStatus.TRUE_POSITIVE,
+                AssessmentStatus.FALSE_POSITIVE, AssessmentStatus.UNDER_REVIEW);
+
+        developerAnalysis = new ComboBox<>("Developer analysis");
+        developerAnalysis.setItems(AppSecData.VulnerabilityStatus.NOT_SET,
+                AppSecData.VulnerabilityStatus.NOT_AFFECTED,
+                AppSecData.VulnerabilityStatus.FALSE_POSITIVE,
+                AppSecData.VulnerabilityStatus.IN_TRIAGE,
+                AppSecData.VulnerabilityStatus.EXPLOITABLE);
+
+        severity = new ComboBox<>("Severity");
+        severity.setItems(SeverityLevel.NONE, SeverityLevel.LOW,
+                SeverityLevel.MEDIUM, SeverityLevel.HIGH,
+                SeverityLevel.CRITICAL);
+
+        riskScore = new ComboBox<>("CVSS score");
+        riskScore.setItems(">= 0", ">=1", ">=2", ">=3", ">=4", ">=5", ">=6",
+                ">=7", ">=8", ">=9", ">=10");
+
+        Component filterBar = buildFilterBar(ecosystem, dependency,
+                vaadinAnalysis, developerAnalysis, severity, riskScore);
+        getMainContent().add(filterBar);
+    }
+
+    private void buildGrid() {
+        grid = new Grid<>();
+        grid.setSelectionMode(Grid.SelectionMode.SINGLE);
+        grid.setColumnReorderingAllowed(true);
```

Comment:
I wonder if at the end making the column order customized really helps: one would expect the order to be save somehow between view refresh and app restarts, but it isn't. Let's discuss this!

Reply:
Removed for now as discussed. Also from the `DependenciesTab`.